app/SharedWebComponents/Pages/Chat.razor.cs

```diff
@@ -1,30 +1,172 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
 namespace SharedWebComponents.Pages;
+using Microsoft.AspNetCore.SignalR.Client;
+using System.Text.Json;
 
-public sealed partial class Chat
+public sealed partial class Chat : IAsyncDisposable
 {
     private string _userQuestion = "";
     private UserQuestion _currentQuestion;
     private string _lastReferenceQuestion = "";
     private bool _isReceivingResponse = false;
+    private bool _useStreaming = true;
```

Comment:
Is it only related to using signalr or not? I think it's not related to create signalr reousrce?

Reply:
Its used to know if we are using streaming mode or not streaming mode. Not related to create signalr resource.

https://github.com/phanthaiduong22/azure-search-openai-demo-csharp/blob/331181f94a1b5b34c9756d7c8d05db39c58f8e0d/app/SharedWebComponents/Pages/Chat.razor#L92